gcloud-java-pubsub/src/main/java/com/google/gcloud/pubsub/PubSub.java

```diff
@@ -0,0 +1,194 @@
+
+import com.google.gcloud.AsyncPage;
+import com.google.gcloud.Page;
+import com.google.gcloud.Service;
+
+import java.io.Serializable;
+import java.util.List;
+import java.util.concurrent.Future;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * An interface for Google Cloud Pub/Sub.
+ *
+ * @see <a href="https://cloud.google.com/pubsub/">Google Cloud Pub/Sub</a>
+ */
+public interface PubSub extends Service<PubSubOptions> {
+
+  final class ListOption implements Serializable {
+
+    private static final long serialVersionUID = 6517442127283383124L;
+
+    private final Option option;
+    private final Object value;
+
+    enum Option {
+      PAGE_SIZE, PAGE_TOKEN
+    }
+
+    private ListOption(Option option, Object value) {
+      this.option = option;
+      this.value = value;
+    }
+
+    Option option() {
+      return option;
+    }
+
+    Object value() {
+      return value;
+    }
+
+    public static ListOption pageSize(int pageSize) {
+      return new ListOption(Option.PAGE_SIZE, pageSize);
+    }
+
+    public static ListOption pageToken(String pageToken) {
+      return new ListOption(Option.PAGE_TOKEN, pageToken);
+    }
+  }
+
+  final class PullOption implements Serializable {
+
+    private static final long serialVersionUID = -5220474819637439937L;
+
+    private final Option option;
+    private final Object value;
+
+    enum Option {
+      RETURN_IMMEDIATELY, MAX_MESSAGES
+    }
+
+    private PullOption(Option option, Object value) {
+      this.option = option;
+      this.value = value;
+    }
+
+    Option option() {
+      return option;
+    }
+
+    Object value() {
+      return value;
+    }
+
+    public static PullOption returnImmediatly() {
+      return new PullOption(Option.RETURN_IMMEDIATELY, true);
+    }
+
+    public static PullOption maxMessages(int maxMessages) {
+      return new PullOption(Option.MAX_MESSAGES, maxMessages);
+    }
+  }
+
+  Topic create(TopicInfo topic);
+
+  Future<Topic> createAsync(TopicInfo topic);
+
+  // null if not found
+  Topic getTopic(String topic);
+
+  Future<Topic> getTopicAsync(String topic);
+
+  // false if not found
+  boolean deleteTopic(String topic);
+
+  Future<Boolean> deleteTopicAsync(String topic);
+
+  Page<Topic> listTopics(ListOption... options);
+
+  Future<AsyncPage<Topic>> listTopicsAsync(ListOption... options);
+
+  String publish(String topic, Message message);
+
+  Future<String> publishAsync(String topic, Message message);
+
+  List<String> publish(String topic, Message message, Message... messages);
+
+  Future<List<String>> publishAsync(String topic, Message message, Message... messages);
+
+  List<String> publish(String topic, Iterable<Message> messages);
+
+  Future<List<String>> publishAsync(String topic, Iterable<Message> messages);
+
+  Subscription create(SubscriptionInfo subscription);
+
+  Future<Subscription> createAsync(SubscriptionInfo subscription);
+
+  // null if not found
+  Subscription getSubscription(String subscription);
```

Comment:
I'm curious about the decision to bundle the publisher and subscriber functionality into the same interface, given that it's separate at the rpc level.